lua/neotest-vstest/dotnet_utils.lua

```diff
@@ -284,26 +284,26 @@ function dotnet_utils.get_solution_projects(solution_path)
 
   local solution_dir = vim.fs.dirname(solution_path)
 
-  local projects = {}
+  local projects = dotnet_utils.projects(solution_path)
 
-  if solution_path then
-    projects = dotnet_utils.projects(solution_path)
-  else
-    logger.info("found no solution file in " .. solution_path)
-    projects = vim.fs.find(function(name, _)
-      return name:match("%.[cf]sproj$")
-    end, { upward = false, type = "file", path = solution_path })
+  local _test_projects = {}
+
+  for _, project in ipairs(projects) do
+    _test_projects[#_test_projects + 1] = function()
+      return dotnet_utils.get_proj_info(project)
+    end
   end
 
+  local all_projects_info = nio.gather(_test_projects)
   local test_projects = {}
-
-  for _, project in ipairs(projects) do
-    local project_info = dotnet_utils.get_proj_info(project)
-    if project_info and project_info.is_test_project then
-      test_projects[#test_projects + 1] = project_info
+  for _, project in ipairs(all_projects_info) do
+    if project and project.is_test_project then
+      test_projects[#test_projects + 1] = project
     end
   end
 
+  local test_projects = nio.gather(_test_projects)
+
```

Comment:
I think this line is not supposed to be here, since the results where filtered just above 😄

Reply:
Riiight😅 Must have been there from when I rebased. Great catch!